public/js/surveyBuilderForm.js

```diff
@@ -0,0 +1,190 @@
+$(function(){
+    var addQuestionButton = $('#add-question')
+
+    addQuestionButton.prop("disabled",true)
+
+    $('#input-selector').change(function()
+    {
+        if ($('#input-selector').val() !== 'text-input' && $('#question-options').length < 1)
+        {
+            addOptionsCreator($('#input-container'))
+
+        } else if($('#input-selector').val() == 'text-input')
```

Comment:
I thought the same initially, as did Mike, but there is a slight distinction: 

The "if" checks if the input selector is being changed from the text type to another type, in which  case the options controller pops up.
The "else if" checks if the new type is text, in which case it removes the options controls. The case where it is being changed from radio to checkbox or vice versa is false in both cases, and hence we neither create nor remove the options controls.